db/document.go

```diff
@@ -811,3 +811,74 @@ func (doc *document) MarshalWithXattr() (data []byte, xdata []byte, err error) {
 
 	return data, xdata, nil
 }
+
+// Equal returns true if both of the given attachment metadata maps are considered the same.
+func (a AttachmentsMeta) Equal(b AttachmentsMeta) bool {
+	if len(a) != len(b) {
+		return false
+	} else if a == nil && b != nil {
+		return false
+	} else if b == nil && a != nil {
+		return false
+	}
+
+	for aAttachmentName, aAttachment := range a {
+		aAttachmentMap, ok := aAttachment.(map[string]interface{})
+		if !ok {
+			return false
+		}
+
+		bAttachment, ok := b[aAttachmentName]
+		if !ok {
+			return false
+		}
+		bAttachmentMap, ok := bAttachment.(map[string]interface{})
+		if !ok {
+			return false
+		}
+
+		// for each attachment property found in a, make sure there's one in b, and it matches
```

Comment:
This doesn't look like it covers the case where there's a property in b that isn't in a.  The underlying question is whether we want to evaluate changes to custom (i.e. non-Couchbase) properties added to _attachments when considering equality.  My feeling is that we shouldn't - I don't think the corner case where a custom property changes while the standard properties (digest, name, etc) don't change is something we need to support.

Based on that, I think we can switch from a generic map comparison to a specific comparison of known properties in the map.